db/repair.go

```diff
@@ -0,0 +1,264 @@
+package db
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	cf "github.com/alphagov/paas-usage-events-collector/cloudfoundry"
+)
+
+// RepairEvents populates START events for applications and services that are
+// missing due to the events being emitted before data collection began (T0).
+//
+// * If an application guid is currently in a running state according to
+//   CloudController, but no events exist, then a START event is created for T0
+// * If a service_instance_guid is currently in a CREATED state according to
+//   CloudController, but no events exist, then a CREATED event is created for T0
+func (pc *PostgresClient) RepairEvents(cfClient cf.Client) (err error) {
```

Comment:
I can't find where this actually gets called from. Where is it used? `ack 'RepairEvents' -i` turned up nothing in the entire repository (other than lines 12 and 19 of this file of course).

Reply:
Good point! 

It was originally executed on startup ... but that was a bit presumptuous. It should be exposed via the API or removed entirely (it's only required as an alternative solution to triggering the cloud controller "pure and reset events" thing).

I'll had a handler for it and document it, then it stays as an option for now (since I found it useful for fixing  a dump of events from prod).

Reply:
Creating a decent `RepairEvents` function that makes the best use of the data we have to create the most accurate fake events it can is a story in itself.